src/integrations/stripe.js

```diff
@@ -0,0 +1,149 @@
+}
+/**
+ * Sets up a payment method (i.e. a card) in stripe under a customer.
+ * @param {*} client
+ * @param {*} formFieldValues the form's field values
+ * @param {*} paymentMethodFieldId the payment method field id
+ * @param {*} stripeConfig the integration config for stripe.  Non-null.
+ * @param {*} stripe the stripe api object.
+ * @param {*} cardElement tthe stripe cardElement (from stripe Elements).
+ */
+export async function setupPaymentMethod(
+  client,
+  formFieldValues,
+  paymentMethodFieldId,
+  stripeConfig,
+  stripe,
+  cardElement
+) {
+  if (!stripeConfig || Object.keys(stripeConfig).length === 0) return null;
+
+  const { intent_secret: intentSecret } = await client.setupPaymentIntent(
+    getStripeCustomerFieldValues(stripeConfig, formFieldValues),
+    paymentMethodFieldId
+  );
+
+  const result = await stripe.confirmCardSetup(intentSecret, {
+    payment_method: {
+      card: cardElement
+      // Not supplying billing details for now but we could later if needed...
+      // billing_details: { email }
+    }
+  });
+  if (result.error) throw new Error(result.error.message);
+  else {
+    return await client.retrievePaymentMethodData(
+      paymentMethodFieldId,
+      result.setupIntent.payment_method
+    );
+  }
+}
+
+export async function setupPaymentMethods(
+  activeStep,
+  client,
+  fieldValues,
+  formattedFields,
+  stripeConfig,
+  cardElementMappings,
+  updateFieldValues,
+  setLoader
+) {
+  // Perform any Stripe payment method setup here if and only if the
+  // payment method field has a complete value
+  const paymentMethodFields = getPaymentMethodFieldsOnStep(activeStep);
+  const haveCompletePaymentMethod = paymentMethodFields.some(
+    (pmf) => formattedFields[pmf.servar.key].value?.complete
+  );
+  if (stripeConfig && haveCompletePaymentMethod) {
+    setLoader();
+    const stripe = await getStripe();
+
+    // Setup each payment method.  It any fails, do not submit.
+    // each payment must be setup one at a time, async/parallel does not work here!
+    for (let i = 0; i < paymentMethodFields.length; i++) {
+      const paymentMethodField = paymentMethodFields[i];
+      if (formattedFields[paymentMethodField.servar.key].value?.complete) {
+        try {
+          const paymentMethodData = await setupPaymentMethod(
+            client,
+            fieldValues,
+            paymentMethodField.servar.key,
+            stripeConfig,
+            stripe,
+            cardElementMappings[paymentMethodField.servar.key]
+          );
+          // Got data for the payment methid field, set it on the field
+          updateFieldValues({
+            [paymentMethodField.servar.key]: paymentMethodData
+          });
+          formattedFields[paymentMethodField.servar.key] = {
+            value: paymentMethodData,
+            type: paymentMethodField.servar.type,
+            displayText: paymentMethodField.servar.name
+          };
+        } catch (e) {
+          throw Promise.resolve({
```

Comment:
you should be able to throw the error directly

Reply:
I simplified here to remove the throw and catch. A simple error return was sufficient.